iroh-net/src/magicsock.rs

```diff
@@ -2000,102 +2000,105 @@ impl Actor {
             .map(|a| a.ip().is_unspecified())
             .unwrap_or(false);
 
-        let LocalAddresses {
-            regular: mut ips,
-            loopback,
-        } = LocalAddresses::new();
-
-        if is_unspecified_v4 || is_unspecified_v6 {
-            if ips.is_empty() && eps.is_empty() {
-                // Only include loopback addresses if we have no
-                // interfaces at all to use as endpoints and don't
-                // have a public IPv4 or IPv6 address. This allows
-                // for localhost testing when you're on a plane and
-                // offline, for example.
-                ips = loopback;
-            }
-            let v4_port = local_addr_v4.and_then(|addr| {
-                if addr.ip().is_unspecified() {
-                    Some(addr.port())
-                } else {
-                    None
+        let msock = self.msock.clone();
+
+        tokio::spawn(async move {
+            let LocalAddresses {
+                regular: mut ips,
+                loopback,
+            } = LocalAddresses::new();
```

Comment:
This can still block the executor this is being spawned on.  Could you wrap the new() call into `tokio::task::spawn_blocking()`?  And also leave a comment that on windows this introduces a 10-20ms of delay so that we know why.

Reply:
I have made the modifications as per your suggestion. I specifically tested the time consumed by the latest version of `netdev::get_interfaces()` under various conditions. The results are as follows: 

```
windows（Lan + wifi off）：min: 9ms, max: 28ms
windows（Lan + wifi connects to mobile hotspot ）：min: 24ms, max: 1174ms

macos (Hackintosh X86): min: 1ms, max: 6ms

linux (7840hs + lan): min: 100us, max: 502us
linux (7840hs + 1lan connect + 3lan disconnect): min: 100us, max: 1092us
linux (2core cpu server + lan): min: 100us, max: 960us
```

The time taken for the function call may be influenced by the platform, the number of network interfaces, and CPU performance (such as on embedded devices or server). Therefore, in the comments, I did not limit my annotations to just Windows. The test results show that the Windows platform indeed takes the most time. Analyzing the source code of `netdev` for the Windows platform confirms that the outcome aligns with expectations.

Reply:
Thanks for these detailed numbers, it's great to see these written down!